File: crates/renderer/src/scene/time.rs
   pub struct Time {
    elapsed_time: f32,
    delta_time: f32,
    frame_count: u64,
}

impl Default for Time {
    fn default() -> Self {
        Self::new()
    }
}

impl Time {
    pub fn new() -> Self {
        Time {
            elapsed_time: 0.0,
            delta_time: 0.0,
            frame_count: 0,
        }
    }

    pub fn update(&mut self, delta_time: f32) {
        self.elapsed_time += delta_time;
        self.delta_time = delta_time;
        self.frame_count += 1;
    }

    pub fn delta_time(&self) -> f32 {
        self.delta_time
    }

    pub fn elapsed_time(&self) -> f32 {
        self.elapsed_time
    }

    pub fn frame_count(&self) -> u64 {
        self.frame_count
    }
}

File: scripting/Ferron/Time.cs
namespace Ferron;

/// <summary>Frame timing, read live from the engine's Time resource.</summary>
public static class Time
{
    /// <summary>Seconds since the last frame.</summary>
    public static float DeltaTime
    {
        // TODO: Native.TimeDelta()
        get => 0f;
    }

    /// <summary>Total elapsed seconds since engine start.</summary>
    public static float Total
    {
        // TODO: Native.TimeTotal()
        get => 0f;
    }

    /// <summary>Frames ticked since engine start; handy for debugging frame-dependent bugs.</summary>
    public static ulong FrameCount
    {
        // TODO: Native.TimeFrameCount()
        get => 0;
    }
}
